src/chainstate/burn/db/sortdb.rs

```diff
@@ -6127,6 +6224,50 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn test_insert_peg_in() {
+        let txid = Txid([0; 32]);
+        let block_height = 123;
+        let vtxindex = 456;
+        let amount = 1337;
+        let recipient = StacksAddress::new(1, Hash160([1u8; 20])).into();
+        let peg_wallet_address =
+            PoxAddress::Addr32(false, address::PoxAddressType32::P2TR, [0; 32]);
+        let burn_header_hash = BurnchainHeaderHash([0x03; 32]);
+
+        let peg_in = PegInOp {
+            recipient,
+            peg_wallet_address,
+            amount,
+
+            txid,
+            vtxindex,
+            block_height,
+            burn_header_hash,
+        };
+
+        let first_burn_hash = BurnchainHeaderHash::from_hex(
+            "0000000000000000000000000000000000000000000000000000000000000000",
+        )
+        .unwrap();
+
+        let epochs = StacksEpoch::unit_test(StacksEpochId::Epoch21, block_height);
+        let mut db =
+            SortitionDB::connect_test_with_epochs(block_height, &first_burn_hash, epochs).unwrap();
+
+        let snapshot = test_append_snapshot(
+            &mut db,
+            BurnchainHeaderHash([0x01; 32]),
+            &vec![BlockstackOperationType::PegIn(peg_in.clone())],
+        );
+
+        let res_peg_ins = SortitionDB::get_peg_in_ops(db.conn(), &burn_header_hash)
+            .expect("Failed to get peg-in ops from sortition DB");
+
+        assert_eq!(res_peg_ins.len(), 1);
+        assert_eq!(res_peg_ins[0].amount, 1337);
```

Comment:
Can you add test coverage to verify that two different burnchain blocks can contain the same transaction (i.e. with the same txid), but that `SortitionDB::get_peg_in_ops()` will return the burnchain block-specific operation metadata for a given burnchain block hash?

Reply:
Done. I extended the test case to submit two peg in operations with different amounts and burn header hashes.